src/EFCore.MySql/Extensions/MySqlDbContextOptionsBuilderExtensions.cs

```diff
@@ -91,27 +91,27 @@ public static class MySqlDbContextOptionsBuilderExtensions
         /// <returns> The options builder so that further configuration can be chained. </returns>
         public static DbContextOptionsBuilder UseMySql(
             [NotNull] this DbContextOptionsBuilder optionsBuilder,
-            [NotNull] string connectionString,
+            [CanBeNull] string connectionString,
             [NotNull] ServerVersion serverVersion,
```

Comment:
Suggest this could be null, if server version is null, it should use the existed conn str to detect version. 

For user experience, we should decrease the must input arguments to make them onboard this framework easier..

Reply:
(I am assuming you suggest `serverVersion` could be nullable, and not `connectionString`.)

I am still not a fan of that. We made the deliberate choice to force users to specify the server version, because functionality will change significantly depending on the database server implementation and version (queries can easily break because of that).

So if someone does not care about the server version their app code depends upon, and because of that they want just to auto detect the server version that is used at runtime, we want them to need to opt-in for that.

(In addition, for production scenarios, you probably want some kind of retry strategy for server version detection, in case the database link is temporarily down for some unexpected reason.)

Reply:
Note that in some scenarios, there really isn't a server to connect to, in order to know which version to target. For example, when generating a migration SQL script, the script is being generated without any database access.